avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -691,9 +691,6 @@ private static Object maybeWrapSchemaless(Schema schema, Object value, String ty
       case STRING:
         if (enhancedSchemaSupport && schema.parameters() != null
             && schema.parameters().containsKey(AVRO_TYPE_ENUM)) {
-          String enumSchemaName = schema.parameters().get(AVRO_TYPE_ENUM);
-          org.apache.avro.Schema cachedEnum = schemaMap.get(enumSchemaName);
-          if (cachedEnum == null) {
```

Comment:
Ah, I see what you're getting at now -- the global check early on in the method should be enough to handle the caching. I think you're right, and actually line 708 also seems redundant, it probably snuck in with the same PR that added the cache check here.